### ISSUE TITLE
Documentation clarification

### ISSUE THREAD

**Issue by yfinkelstein** [lewissbaker/cppcoro#44]
in task.hpp there is 
```
/// \brief
	/// A lazy task represents an asynchronous operation that is not started
	/// until it is first awaited.
	///
	/// When you call a coroutine that returns a task, the coroutine
	/// simply captures any passed parameters and returns exeuction to the
	/// caller. Execution of the coroutine body does not start until the
	/// coroutine is first co_await'ed.
	///
	/// Comparison with task<T>
	/// -----------------------
	/// The lazy task has lower overhead than cppcoro::task<T> as it does not
	/// require the use of atomic operations to synchronise potential races
	/// between the awaiting coroutine suspending and the coroutine completing.
	///
	/// The awaiting coroutine is suspended prior to the task being started
	/// which means that when the task completes it can unconditionally
	/// resume the awaiter.
	///
	/// One limitation of this approach is that if the task completes
	/// synchronously then, unless the compiler is able to perform tail-calls,
	/// the awaiting coroutine will be resumed inside a nested stack-frame.
	/// This call lead to stack-overflow if long chains of tasks complete
	/// synchronously.
	///
	/// The task<T> type does not have this issue as the awaiting coroutine is
	/// not suspended in the case that the task completes synchronously.
	template<typename T = void>
	class task

```
The section `Comparison with task<T>` is not clear: what is being compared with `task<T>`? This IS the `task<T>` definition after all.

Btw, what is the best venue to ask similar questions in the future?

**Comment by lewissbaker** [comment]
Github issues is the beat place for raising issues like this. You can also generally find people who should be able to answer questions on cpplang.slack.com #coroutines channel.

As for the confusing docs, it's a remnant from when I used to have two task classes. A `laz_task` and an eager `task`. I removed the eager task and renamed `lazy_task` to `task`. Must have missed this bit of doco.

I'll fix it up.

**Comment by lewissbaker** [comment]
Fixed in 03f8027c167ffb9cc5f1a33373290b7a0c8b7b41.
